test/test_no_fake_tensors.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env pytest
+import functools
+from unittest.mock import patch
+
+import torchdynamo
+
+from . import test_functions
+from . import test_misc
+from . import test_modules
+from . import test_repros
+from . import test_unspec
+
+
+def make_no_fake_fn(fn):
```

Comment:
I know I mentioned making a separate file for this, and I think we need to for the sake of pytest and our sanity, but I think there's a good amount of logic here we can share with the dynamic shape test this file is based on. 

Maybe we can pull out this function and the one below it into a shared module?

Reply:
Done, way cleaner now, and supports an arbitrary number of patches 👯

Reply:
Very nice :) it LGTM and our recent config changes should keep that patch valid.